Development/nmos/system_resources.cpp

```diff
@@ -35,13 +69,32 @@ namespace nmos
         const auto& is04 = nmos::fields::is04(data);
         const auto& ptp = nmos::fields::ptp(data);
 
+        auto global_data = value_of({
+            { nmos::fields::registration_heartbeat_interval, nmos::fields::heartbeat_interval(is04) },
+            { nmos::fields::ptp_announce_receipt_timeout, nmos::fields::announce_receipt_timeout(ptp) },
+            { nmos::fields::ptp_domain_number, nmos::fields::domain_number(ptp) },
+            { nmos::fields::tags, data.at(nmos::fields::tags) }
+        });
+
+        if (data.has_field(nmos::fields::syslog))
+        {
+            auto syslog = value_of({
+                { nmos::fields::syslog, nmos::fields::syslog(data) }
+            });
+            web::json::merge_patch(global_data, syslog, true);
```

Comment:
If I understand the intent correctly, perhaps this can more succinctly and efficiently be written as:
```C++
global_data[nmos::fields::syslog] = nmos::fields::syslog(data);
```
or if above suggestion is applied as:
```C++
global_data[nmos::fields::syslog] = syslog;
```

Reply:
On the other hand, this code in `parse_system_global_data` is incompatible with `make_system_global_data`. These two functions should be able to roundtrip their args/return values. This code produces an `nmos::settings` with a `syslog` top-level property which is an object, whereas the `make...` function expects the two top-level properties `system_syslog_host_name` and  `system_syslog_port`.

Probably all the pairs of `make...` and `parse...` functions ought to have had unit tests that demonstrate that. Mea culpa!

Reply:
yes, you are right, make & parse functions should be matching. Fixed, and the unit test has now been added, ready for Joan to review.

Reply:
👍